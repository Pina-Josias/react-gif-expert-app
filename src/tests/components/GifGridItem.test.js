import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Pruebas sobre GifGridItem.js", () => {
  const title = "Gif de prueba";
  const url = "https://i.gifer.com/9oXG.gif";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe mostrar el componente correctamente", async () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe el componente tener un párrafo con el título "Gif de prueba"', () => {
    const p = wrapper.find("p");
    expect(p.text()).toBe(title);
  });

  test("debe tener la imagen al url y alt de los props", () => {
    const img = wrapper.find("img");
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });
  test("debe el componente tener un div con animate__fadeIn", () => {
    const div = wrapper.find("div");
    expect(div.props().className).toContain("animate__fadeIn");
  });
});
