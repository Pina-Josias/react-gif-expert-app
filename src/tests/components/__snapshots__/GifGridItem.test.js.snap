// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifGridItem.js Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="Gif de prueba"
    src="https://i.gifer.com/9oXG.gif"
  />
  <p>
    Gif de prueba
  </p>
</div>
`;
